### ISSUE TITLE
The from address on the confirmation email for updating email in Admin Settings (Support/Newsletter) is incorrect

### ISSUE THREAD

**Issue by shubhank008** [TryGhost/Ghost#12802]
### Issue Summary

Currently, using Ghost v4.0, if you try to update the emails in your Admin Settings of Ghost under Members->Email and try to update either support or newsletter, they are set to noreply@domain.tld  
They should ideally use the "from" address if set in your config's mail part.  

### To Reproduce

The bigger problem is not this, its the fact that if you set custom SMTP and even "from" address in your config's mail part, and try to update the emails given in your Admin Area above, Ghost ignores whatever is set in your config and still uses noreply@domain.tld or no-reply@domain.tld which creates issue with SMTP config as when Ghost tries to send such mails (confirmation mails), the SMTP server will reject it as the "from" mail is different from the actual userid/smtp emailid.  

Any other info e.g. Why do you consider this to be a bug? What did you expect to happen instead?

This is how postfix (SMTP) logs it:  
```
NOQUEUE: reject: RCPT from unknown[xx.xx.xx.xx]: 553 5.7.1 <noreply@dly.in>: Sender address rejected: not owned by user user1@dly.in; from=<noreply@dly.in> to=<user1@dly.in> proto=ESMTP helo=<[127.0.0.1]>   
```  

The logic for this part (and how I fixed it) lies here:  
`core/server/services/members/settings.js`  

This is how Ghost currently does it  
```
const [,toDomain] = email.split('@');
        let fromEmail = `noreply@${toDomain}`;
        if (fromEmail === email) {
            fromEmail = `no-reply@${toDomain}`;
        }
```

Ideally it should atleast check if a from address exists in your config.mail or use the smtp id instead.  This just limits the usage of mail and SMTP in general to a very handful of providers.  

I had to fix it by overriding fromEmail to use email itself and then I was able to update the mail in my Admin:  
```
let fromEmail = `noreply@${toDomain}`;
        if (fromEmail === email) {
            fromEmail = `no-reply@${toDomain}`;
        }
        fromEmail = email;
```

### Technical details:

* Ghost Version:  v4.0
* Node Version:
* Browser/OS:  Chrome
* Database:  MySQL

**Comment by ErisDS** [comment]
Hey @shubhank008, I hear that you are frustrated but I have to be honest. I've read this 3x and I don't quite follow what you are saying.

Please can you go back to the issue template we provided and supply a step by step reproduction case demonstrating the problem so that we can try to reproduce it?

**Comment by shubhank008** [comment]
> Hey @shubhank008, I hear that you are frustrated but I have to be honest. I've read this 3x and I don't quite follow what you are saying.
> 
> Please can you go back to the issue template we provided and supply a step by step reproduction case demonstrating the problem so that we can try to reproduce it?

Woops sorry, edited my initial issue above.  I am not frustrated as I found a way to fix it (albeit using a solution provided by another user in ghost forums facing same issue and pointing the problem in settings.js).  

To try and explain it a bit more:  

1. In ghost admin panel, under Members settings, if we try to update the email ids used for Newsletters and Support, they are set to noreply@yourdomain.com when ideally they should be set to what we have set as the emailid in our config.json file (the mail.from config field).  
2. So even a user sets a "from" email ID to be used in config.json, its not used by ghost for support and newsletters as they continue to use that noreply id which is set through the "core/server/services/members/settings.js" file.  
3. Now you would think, the fix is as simple as just updating the default email id in Admin panel from "noreply" to youremail@domain.com but here is the problem, when you try to make that update, Ghost sends a confirmation mail to that new email id you are trying to set and it tries to do so using the current email id that is set in Members area, which is "noreply@"  
4. The problem that follows after it is there is no such id "noreply@" if using SMTP and as such, the confirmation mail is never sent and instead the SMTP server responds with a error shows in my initial post stating this email ID is not correct and different from your SMTP details (i.e. in our SMTP we are using youremail@ but Ghost sends that mail with noreply@)  
5. The proper way to do this should have been to honour and use the actual ID that is set in the config.json file (config.mail.from)  as you can see from my temporary solution to bypass that noreply check, the moment we do that and use actual emailID directly, it works.

**Comment by mascali33** [comment]
This is still a problem, using Ghost 4.18 with a custom SMTP server, ghost is not using the from adress specified in config and uses noreply@domain.tld instead of mail.user@mailhost.tld like said in config.
Ghost should only use the noreply@domain.tld when no "mail__from" value is set

**Comment by mascali33** [comment]
Its also in the docs [https://ghost.org/docs/config/#from-address](https://ghost.org/docs/config/#from-address)

Ping @ErisDS

**Comment by mascali33** [comment]
> > Its also in the docs https://ghost.org/docs/config/#from-address
> > Ping @ErisDS
> 
> Quite disappointed has been months since it was reported and in so much details and it was a 2 line code fix (well rather override the config) but the bug its still there. Makes me wonder do not a whole lot ghost users use SMTP or check/test their mails ?

Sometime it seems they don't see the issues :\
Can you try to open a pull request fix and reference this issue? Maybe they will see the problem :\

**Comment by tryfan** [comment]
This is still an issue.

Supplying `mail__from` as an env var does not change the email from address in a container.  Just tested on 4.36.1.

**Comment by ErminMerdanovic** [comment]
I encounter the issue today as well. Still an issue

**Comment by mgk1231** [comment]
This is still a valid bug. As of today this issue is not yet resolved

**Comment by Antonyrahul** [comment]
> > Facing this issue. Is there any temporary workaround for now. I am using aws smtp
> 
> You can use the temporary codefix I listed in my original report (the very top post)

I did that fix and restaeted the server and the mail is still being sent from noreply. Is there something i ma missing . I have cleared the cache and tried too but it does not work

**Comment by marnixhoh** [comment]
Still running into the same issue. Related:
#15191
#15222

So many people must be impacted by this...

**Comment by shubhank008** [comment]
Anyone still facing this god forsaken issue, my fix still works except the location/file where you need to make the change has changed in latest ghost (v5) from 
`core/server/services/members/settings.js`
to
`core/server/services/settings/settings-bread-service.js`

And don't forget to set the `from` field in your config's mail section

```
"mail": {
    "from": "'Name' <something@domain.tld>",
    "transport": "SMTP",
    "options": {
      "auth": {
        "user": "something@domain.tld",
        "pass": "123456"
      },
      "host": "mail.domain.tld",
      "port": 587
    }
  },
```

> 
> The logic for this part (and how I fixed it) lies here: `core/server/services/members/settings.js`
> 
> This is how Ghost currently does it
> 
> ```
> const [,toDomain] = email.split('@');
>         let fromEmail = `noreply@${toDomain}`;
>         if (fromEmail === email) {
>             fromEmail = `no-reply@${toDomain}`;
>         }
> ```
> 
> Ideally it should atleast check if a from address exists in your config.mail or use the smtp id instead. This just limits the usage of mail and SMTP in general to a very handful of providers.
> 
> I had to fix it by overriding fromEmail to use email itself and then I was able to update the mail in my Admin:
> 
> ```
> let fromEmail = `noreply@${toDomain}`;
>         if (fromEmail === email) {
>             fromEmail = `no-reply@${toDomain}`;
>         }
>         fromEmail = email;
> ```

**Comment by marnixhoh** [comment]
I had this issue too, and I found the solution, but forgot to post it here...

Anyway, after having created multiple issues here on Github and on the Ghost Forum, someone from their staff gave the answer on [this forum thread](https://forum.ghost.org/t/mail-from-option-shows-no-effect/31857/24).

I hope it helps you and others :)

**Comment by ErisDS** [comment]
@marnixhoh I think you are confusing a different issue. @shubhank008 If you have a fix for the original issue, please raise a PR.

**Comment by lnlclay** [comment]
I thought my postfix box was broken until I find this issue here, 18 months later still not fixed.

**Comment by lvnilesh** [comment]
Hope this PR gets merged soon https://github.com/TryGhost/Ghost/pull/15989

**Comment by ph1958** [comment]
It seems Ghost core has zero interest in the issues that are collected then shelved in this Issue.

Interesting thing is GhostPro behaves similarly but they offer a fix. Upgrade to "business" plan and pay $$$$$
"though this does require a Business plan, alongside a $50/month add-on for support"

**Comment by ErisDS** [comment]
Please see my responses over on the PR here: https://github.com/TryGhost/Ghost/pull/15989#issuecomment-1766567698

TL;DR we have a piece of work to change how from addresses work in a holistic way, that is scheduled to be started on immediately.

**Comment by ph1958** [comment]
> Please see my responses over on the PR here: [#15989 (comment)](https://github.com/TryGhost/Ghost/pull/15989#issuecomment-1766567698)
> 
> TL;DR we have a piece of work to **_change how from addresses work in a holistic way_**, that is scheduled to be started on immediately.

@ErisDS Thanks for getting the ball rolling on fixing the various issues with how ghost deals with mail.
Please provide specifics explaining what **_change how from addresses work in a holistic way_** means.

Will we have appropriate control over setting email from addresses consistent with Mailgun and others?
Will the sent via nonsense be a thing of the past?
Will Ghost begin setting mail envelopes consistent with industry standards?

It would set some minds at ease if the plan was actually shared with folks rather than springing the result on us. We all saw how the surprise approach works vis-a-vis the admin settings change debacle.

**Comment by henrywithu** [comment]
The issue remains.
For "Support email address", why is it fixed? (I'm using docker)
It's like being forced to send verification email from noreply@example.com, and it's hardcoded

**Comment by ivosabev** [comment]
How is it possible this issue to not be fixed for 2 years?! I am very close from moving to a different platform, because of something so simple.

**Comment by henrywithu** [comment]
> How is it possible this issue to not be fixed for 2 years?! I am very close from moving to a different platform, because of something so simple.

Actually I solved it (if you really want to use personal email to send newsletter or support), by setting email alias like noreply@example.com and no-reply@example.com in your SMTP service provider.
Anyway it's NOT recommended to use personal email to do so, because there're chances that the email is considered as spam.

**Comment by JohnONolan** [comment]
Google and Yahoo are introducing major changes to enforce DMARC in the next 2 months. The core team is currently doing a larger piece of work to support DMARC alignment in email sending, which overlaps-with and will also resolve this issue.

These changes will be released in January 2024.

**Comment by adamu** [comment]
I just set up Ghost in Docker, made sure I configured the `mail__from` environment variable, tried to made a test subscription request and:

```
ghost  | [2024-01-27 14:28:25] WARN Invalid from address used for sending emails: noreply@localhost
ghost  | [2024-01-27 14:28:26] ERROR "POST /members/api/send-magic-link/" 500 787ms
ghost  |
ghost  | Failed to send email. Reason: Message failed: 554 Message rejected: Email address is not verified. The following identities failed the check in region AP-NORTHEAST-1: noreply@localhost.
ghost  |
ghost  | "Please see https://ghost.org/docs/config/#mail for instructions on configuring email."
```

**Comment by SimonBackx** [comment]
🎉 This issue and related issues have been fixed in v5.78.0.

**Changes in Ghost v5.78.0**
* The verification step is dropped when changing the newsletter or support address. You can choose any support and newsletter email address in the UI as long as your SMTP-server / Mailgun account can send from it.
* The `mail.from` config will be the default address for all outgoing emails:
    * Staff notification emails no longer use the made up `ghost` `@` `domain` email address
    * Newsletters no longer default to `noreply` `@` `domain` if no from address is set
    * Member related emails (signin/signup/comment notifications...) will continue to be send from the chosen support address (Portal settings → Account page), but will now default to the `mail.from` config instead of `noreply` `@` `domain` if no support address is set.

**What do you need to do**
* Update Ghost to v5.78.0
* You don’t need to make any changes to your existing configuration if you already set the `mail.from` config. If no `mail.from` config is set, we’ll continue to default to `noreply@domain` everywhere, just like before.

---

We've updated [the documentation](https://ghost.org/docs/config#mail) about the config, and recommend to check out [our post on the forum](https://forum.ghost.org/t/new-email-requirements-in-2024-what-you-need-to-know/44572) about DMARC alignment and the new requirements from Google and Yahoo.

**Comment by sonyarianto** [comment]
How about create config for mail.support in addition to mail.from, so mail.from is only for sender from address, but the mail.support can be any e-mail address that intended for support e-mail, but this support e-mail will not treated as sender e-mail address when Ghost sending communication to user/member.

I think this will makes the configuration more flexible and accomodate several use cases.

**Comment by MFYDev** [comment]
3 years later and this issue is still happening
![](https://private-user-images.githubusercontent.com/53750381/380573643-28604942-d826-4055-aa9d-09389b718d87.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwODg0OTcsIm5iZiI6MTczMDA4ODE5NywicGF0aCI6Ii81Mzc1MDM4MS8zODA1NzM2NDMtMjg2MDQ5NDItZDgyNi00MDU1LWFhOWQtMDkzODliNzE4ZDg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI4VDA0MDMxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1ZGM1NThjYjI4Y2E2YjZlYzQyNjBlYzA4ODQzZjVkNmQyYjQ2OThiMDdhZjdmYjJkMzVhZTMyMjk3NGVmNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ohu852vq6xZPHICOKBcM0bQ2NWkb5Bk5YZqcrPuahnM)

**Comment by laravieira** [comment]
Again, spent a long time trying to figure out why it wasn't sending the e-mails, till I got here and tried to add <noreply@...> as an alias to my email provider, then it worked.

The `mail.from` configuration is <dev@...>, but it sends 2FA and reset password using <noreply@...> as the from.

The error I was getting is nothing helpful:
```
Failed to send email. Please check your site configuration and try again.

"Please see https://ghost.org/docs/config/#mail for instructions on configuring email."

Error ID:
    940b45e0-3dbf-11f0-b00e-8f92039f57ae

Error Code: 
    EMESSAGE

----------------------------------------

EmailError: Failed to send email. Please check your site configuration and try again.
    at Object.sendAuthCodeToUser (/var/lib/ghost/versions/5.120.4/core/server/services/auth/session/session-service.js:284:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.createSession (/var/lib/ghost/versions/5.120.4/core/server/services/auth/session/middleware.js:16:17)
```

**Comment by Polonco** [comment]
> Again, spent a long time trying to figure out why it wasn't sending the e-mails, till I got here and tried to add <noreply@...> as an alias on my email provider, then it worked.
> 
> The `mail.from` configuration is <dev@...>, but it sends 2FA and reset password using <noreply@...> as the from.
> 
> The error I was getting is nothing helpful:
> 
> ```
> Failed to send email. Please check your site configuration and try again.
> 
> "Please see https://ghost.org/docs/config/#mail for instructions on configuring email."
> 
> Error ID:
>     940b45e0-3dbf-11f0-b00e-8f92039f57ae
> 
> Error Code: 
>     EMESSAGE
> 
> ----------------------------------------
> 
> EmailError: Failed to send email. Please check your site configuration and try again.
>     at Object.sendAuthCodeToUser (/var/lib/ghost/versions/5.120.4/core/server/services/auth/session/session-service.js:284:19)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async Object.createSession (/var/lib/ghost/versions/5.120.4/core/server/services/auth/session/middleware.js:16:17)
> ```

same problem here. I can't even create the admin user because it can't send the first email. Following you can see the relevant part of the docker compose I'm using. I tried to specify the from email in 3 different ways, and none of them works. The email is still sent from noreply@edited.org

```
version: "3.8"
services:
  ghost:
    image: ghost:6-alpine
    restart: unless-stopped
    pull_policy: always
    environment:
      database__client: mysql
      database__connection__host: db
      database__connection__user: edited
      database__connection__password: edited
      database__connection__database: ghost
      url: http://${GHOST_HOST}
      mail__transport: SMTP
      mail__options__from: "edited@edited.org"
      from: "edit@edited.org"
      mail__from: "edited@edited.org"
      mail__options__host: smtp.edited.org
      mail__options__port: 587
      mail__options__secure: "false"
      mail__options__auth__user: edited@edited.org
      mail__options__auth__pass: edited
```

**Comment by planetahuevo** [comment]
Technically 5 years and counting. 
https://github.com/TryGhost/Ghost/issues/12152
